src/systems/waypoint-system.js

```diff
@@ -323,7 +329,24 @@ export class WaypointSystem {
       this.waitOneTick = false;
       return;
     }
+
+    const hashUpdated = window.location.hash !== "" && this.previousWaypointHash !== window.location.hash;
+
+    if (hashUpdated && this.initialSpawnHappened) {
+      this.previousWaypointHash = window.location.hash;
+      const spawnPointName = window.location.hash.replace("#", "");
+      const spawnPoint = this.ready.find(c => c.el.className === spawnPointName);
+      if (spawnPoint) {
+        this.occupySpawnPoint(spawnPoint);
+        this.currentMoveToSpawn = null;
+        this.currentMoveToSpawnResolve = null;
+        this.nextMoveToSpawn = null;
+        this.nextMoveToSpawnResolve = null;
+      }
+    }
+
     if (!this.currentMoveToSpawn && this.nextMoveToSpawn) {
+      this.initialSpawnHappened = true;
```

Comment:
This is when the initial spawn begins, but it might finish on a later frame. Here might be a more appropriate place: https://github.com/mozilla/hubs/blob/40173b2e6a90ed7825bcfa0e6f038b384fbf07c7/src/systems/waypoint-system.js#L375

Reply:
Done! thanks